### ISSUE TITLE
Issue with nvm and .nvmrc?

### ISSUE THREAD

**Issue by ryexley** [sindresorhus/pure#278]
I use nvm for node development, and have it setup to detect `.nvmrc` files and call `nvm use` when it finds one. I've been having an issue where I've found that the prompt seems to get updated to include the output from the `nvm use` command in the prompt when this runs, which results in a prompt with a bunch of stuff that I don't want. It seems to only happen with the Pure prompt, and no others that I've tried. [I initially inquired about this as a potential issue with nvm](https://github.com/creationix/nvm/issues/1388#issue-201869493), but after some discussion, it seems like the problem was narrowed down to the use of Pure prompt? Perhaps there's something else I'm missing?

I should also note that it seems to only happen in directories _other than_ the root directory. I have a `.nvmrc` file in my profile directory to set the version of node I wish to use by default, and when I just run `cd` to go to that directory, I don't experience the issue. But whenever I switch to any other directory that has a `.nvmrc` file in it, I see the issue.

### General information

- Pure version: I've tried the version that ships with prezto, and the npm version
- ZSH version: 5.3.1 (x86_64-apple-darwin15.6.0)
- Terminal program & version: I have experienced this with both Hyper v1.1.0.1408 and iTerm v3.0.13
- Operating system: OS X 10.11.6
- ZSH framework: prezto

I have:
- [ x ] Tested with another terminal program and can reproduce the issue: 
- [ x ] Followed the [Integration](https://github.com/sindresorhus/pure#integration) instructions for my framework

### Problem description
<img width="942" alt="screen shot 2017-01-19 at 8 45 16 am" src="https://cloud.githubusercontent.com/assets/173844/22118312/3b7c4218-de45-11e6-8db5-09f9368a950b.png">

### Reproduction steps

1. Just switch to any directory that contains a `.nvmrc` file

### My `.zshrc`:
I currently have the pure prompt initialization commented out, as I initialize it in my `.zpreztorc` file instead

```shell
# Source Prezto.
if [[ -s "${ZDOTDIR:-$HOME}/.dotfiles/.zprezto/init.zsh" ]]; then
  source "${ZDOTDIR:-$HOME}/.dotfiles/.zprezto/init.zsh"
fi

# pure prompt
# autoload -U promptinit && promptinit
# prompt pure

# Load NVM into a shell session *as a function*
if [ -f "$HOME/.nvm/nvm.sh" ]; then
   [[ -s "$HOME/.nvm/nvm.sh" ]] && source "$HOME/.nvm/nvm.sh"

   # Explicitly set the node.js version to use with NVM if no .nvmrc is present (which it should be)
   # nvm use default
fi

# autoload new node version when changing into a directory in which an .nvmrc file is found
autoload -U add-zsh-hook
load-nvmrc() {
  local node_version="$(nvm version)"
  local nvmrc_path="$(nvm_find_nvmrc)"

  if [ -n "$nvmrc_path" ]; then
    local nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")

    if [ "$nvmrc_node_version" = "N/A" ]; then
      nvm install
    elif [ "$nvmrc_node_version" != "$node_version" ]; then
      nvm use
    fi
  elif [ "$node_version" != "$(nvm version default)" ]; then
    echo "Reverting to nvm default version"
    nvm use default
  fi
}
add-zsh-hook chpwd load-nvmrc
load-nvmrc
```

**Comment by mafredri** [comment]
Please try updating Pure, the version of Pure in prezto is a bit old and has an old version of `zsh-async`. The issue should be fixed in more recent versions of `zsh-async`.

What (I believe) you're seeing is your `chpwd` hook being active inside the async worker, and instead of just printing the git dirty status, the `chpwd` output is printed as well.

**Comment by mafredri** [comment]
And, to clarify, I mean update to the latest `master`. The current release (1.4.1) is also old and in due for an update :-).

**Comment by ryexley** [comment]
I will fork the repo and give that a shot. Will report back and close if that fixes the issue. Thanks.

**Comment by ryexley** [comment]
OK, so...I forked the repo, added the symlinks to `pure.zsh` and `async.zsh` and updated my `.zshrc` file as described in the README. I still have the same issue. Turned out, I hadn't linked them to the right place for prezto to pick them up at first. I linked them again into the prezto prompt modules folder, and now it seems to be working as expected. Closing the issue. Thanks for pointing me in the right direction, I appreciate your time and help.
